### ISSUE TITLE
Enabled Metrics not applied - Terraform thinks its applied.

### ISSUE THREAD

**Issue by urjitbhatia** [hashicorp/terraform#7764]
### Affected Resource(s)
Please list the resources as a list, for example:
- aws_autoscalinggroup

### Terraform Configuration Files
Here is my ASG resource block
```
resource "aws_autoscaling_group" "asg" {
  count                = "${var.num_environments}"
  availability_zones   = ["${split(",", var.availability_zones)}"]
  name                 = "${var.service_name}-${lookup(var.environment_names, count.index)}"
  max_size             = "${lookup(var.asg_max, lookup(var.environment_names, count.index))}"
  min_size             = "${lookup(var.asg_min, lookup(var.environment_names, count.index))}"
  launch_configuration = "${element(aws_launch_configuration.lc.*.name, count.index)}"

  enabled_metrics = ["GroupTerminatingInstances", "GroupMaxSize", "GroupDesiredCapacity", "GroupPendingInstances", "GroupInServiceInstances", "GroupMinSize", "GroupTotalInstances"]

  vpc_zone_identifier = ["${split(",", var.subnet_ids)}"]

  lifecycle {
    create_before_destroy = true
  }

  tag {
    key                 = "Name"
    value               = "${var.service_name}-${lookup(var.environment_names, count.index)}"
    propagate_at_launch = "true"
  }
}
```

### Debug Output
https://gist.github.com/urjitbhatia/0b3e64c0b7a886c23b3d1890f9cf612b

### Expected Behavior
When I ran apply, it detected the diff in enabled metrics and "applied" it. Group should now have enabled metrics.

### Actual Behavior
Both the AWS console and cli show enabled metrics is empty and metrics were in-fact not enabled.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply` with a group without enabled_metrics.
2. `terraform apply` with a group with enabled_metrics.

**Comment by stack72** [comment]
Hi @urjitbhatia 

My apologies for the issue here - hopefully you haven't spent too much time on it :( The good news, this has been fixed as of yesterday (#7698). The bad news is that it will not be available until the 0.7 release which we hope will happen soon

Hope this is ok

Thanks

Paul

**Comment by urjitbhatia** [comment]
@stack72 That was a super speedy reply! No worried, I ended up doing it from the aws cli but wanted to let you guys know about the bug anyways.

Thanks for the fix.
